velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1041,4 +1041,33 @@ struct ParseDateTimeFunction {
   }
 };
 
+template <typename T>
+struct TimeZoneHourFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      int64_t& result,
+      const arg_type<TimestampWithTimezone>& timestampWithTimezone) {
+    // Convert timestampWithTimezone to a timestamp representing the moment at
+    // the zone in timestampWithTimezone.
+    Timestamp timeStamp = this->toTimestamp(timestampWithTimezone);
+
+    // Get timestamp in seconds
+    time_t epoch = timeStamp.getSeconds();
+
+    // Get the given timezone name
+    auto timezone =
+        util::getTimeZoneName(*timestampWithTimezone.template at<1>());
+    // Create tm struct with given timezone
+    setenv("TZ", timezone.c_str(), 1);
```

Comment:
I have some reservations about setting the TZ and then consequently using localtime .  Is this safe for multithreaded code ? Whats the cost of setenv ? 
I presume localtime is call to stdlib, is that portable ?